store/sqlstore/store.go

```diff
@@ -146,8 +154,30 @@ func New(settings model.SqlSettings, metrics einterfaces.MetricsInterface) *SqlS
 		settings:  &settings,
 	}
 
+	// We need to tell the MySQL driver that we want to use multiStatements
+	// in order to make migrations work.
+	if store.DriverName() == model.DATABASE_DRIVER_MYSQL {
+		u, err := url.Parse(*settings.DataSource)
+		if err != nil {
+			mlog.Critical("Invalid database url found", mlog.Err(err))
+			time.Sleep(time.Second)
+			os.Exit(ExitGenericFailure)
+		}
+		q := u.Query()
+		q.Set("multiStatements", "true")
```

Comment:
Just checking, IIUC, the `multiStatements` option will be applied for the entire runtime of the application right? Because the migrations use the same connection as the rest of the app, and it won't be _only_ used as a CLI app.

Since we are already calling `setupConnection` inside the migrate method, I am wondering if it's possible to create a separate connection with `multiStatements` to true and then run migrations for it for MySQL?

If we don't want to go to such complexity, I'd just suggest to let the option be added from the DSN itself. So this will need to be part of the release notes and something the customer has to do. IMO, it's better than us changing it under the hood, which might be surprising and not something the customer would expect.

Reply:
Since this is a migrations-only flag and also is a client-owned flag, I prefer to move that into the migrations connection and then tear it down when we are done with executing the migrations. Will amend.